File: MarxUp/MarxUp/ViewControllers/ToolboxItemsViewControllers/LineWidthViewController.m
//
//  LineWidthViewController.m
//  MarxUp
//
//  Created by Ognyanka Boneva on 2.10.18.
//  Copyright © 2018 Ognyanka Boneva. All rights reserved.
//

#import "LineWidthViewController.h"

@interface LineWidthViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *lineImageView;
@property (weak, nonatomic) IBOutlet UISlider *widthSlider;

@end

@implementation LineWidthViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setImageViewLineWidth:self.widthSlider.value];
}

- (IBAction)onSlide:(id)sender {
    UISlider *slider = (UISlider *)sender;
    [self setImageViewLineWidth:slider.value];
    [self.toolboxItemDelegate didChooseLineWidth:slider.value];
}

- (void)setImageViewLineWidth:(CGFloat)width {
    UIGraphicsBeginImageContext(self.lineImageView.frame.size);
    CGContextSetLineWidth(UIGraphicsGetCurrentContext(), width);
    CGContextSetRGBStrokeColor(UIGraphicsGetCurrentContext(), 0.0, 0.0, 0.0, 1.0);
    CGContextMoveToPoint(UIGraphicsGetCurrentContext(), 0, self.lineImageView.frame.origin.y / 2);
    CGContextAddLineToPoint(UIGraphicsGetCurrentContext(), self.lineImageView.frame.size.width, self.lineImageView.frame.origin.y / 2);
    CGContextStrokePath(UIGraphicsGetCurrentContext());
    self.lineImageView.image  = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
}

@end

File: MarxUpUp/MarxUpUp/ViewContollers/ToolboxItemViewControllers/LineWidthViewController.swift
//
//  LineWidthViewController.swift
//  MarxUpUp
//
//  Created by Ognyanka Boneva on 13.11.18.
//  Copyright © 2018 Ognyanka Boneva. All rights reserved.
//

import UIKit

class LineWidthViewController: UIViewController {

    @IBOutlet weak var widthSlider: UISlider!
    @IBOutlet weak var widthImageView: UIImageView!
    weak var toolboxItemDelegate: ToolboxItemDelegate?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = UIColor.clear
        setLineWidth(width: widthSlider.value)
    }
    
    @IBAction func onSlide(_ sender: UISlider) {
        setLineWidth(width: sender.value)
        toolboxItemDelegate?.didChoose(lineWidth: sender.value)
    }
    
    private func setLineWidth(width: Float) {
        UIGraphicsBeginImageContext(widthImageView.frame.size)
        
        let path = UIBezierPath()
        path.lineWidth = CGFloat(width)
        path.move(to: CGPoint(x: 0, y: widthImageView.frame.height / 2))
        path.addLine(to: CGPoint(x: widthImageView.frame.width, y: widthImageView.frame.height / 2))
        UIColor.black.setStroke()
        path.stroke()
        
        widthImageView.image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext();
    }
}
